halyard-web/src/main/java/com/netflix/spinnaker/halyard/controllers/v1/AccountController.java

```diff
@@ -97,43 +88,40 @@
     builder.setUpdate(() -> accountService.addAccount(deploymentName, providerName, account));
     builder.setFieldOptionsResponse(() -> accountService
         .getAccountOptions(deploymentName, providerName, accountName, fieldName));
-    builder.setSeverity(severity);
+    builder.setSeverity(validationSettings.getSeverity());
 
     return DaemonTaskHandler.submitTask(builder::build, "Get " + fieldName + " options");
   }
 
   @RequestMapping(value = "/account/{accountName:.+}/options", method = RequestMethod.PUT)
-  DaemonTask<Halconfig, List<String>> existingAccountOptions(
-      @PathVariable String deploymentName,
+  DaemonTask<Halconfig, List<String>> existingAccountOptions(@PathVariable String deploymentName,
       @PathVariable String providerName,
       @PathVariable String accountName,
-      @RequestParam(required = false, defaultValue = DefaultValidationSettings.severity) Severity severity,
+      @ModelAttribute ValidationSettings validationSettings,
```

Comment:
How are ModelAttributes passed to the controller from an HTTP perspective?

Reply:
They are initialized in a function annotated as `@ModelAttribute` in a class annotated as `@ControllerAdvice`; in this case [here](https://github.com/spinnaker/halyard/blob/fb75291ed92206a0cec4ce7a07ed902c4b94569a/halyard-web/src/main/java/com/netflix/spinnaker/halyard/models/v1/ValidationSettingsHandler.java#L25)

Reply:
Yes, the `@ModelAttribute`-annotated method runs before the handler methods to populate model attributes, which can then be referenced in the handler method.  In this case, it's to encapsulate the common parameters for all handlers into an object rather than have them separately in all handlers.

Reply:
And in this case the `@ModelAttribute` method is just reading from the request params, so the interface is the same for the client.

Reply:
Nice :+1: